### PR TITLE
Refactor of conv fp32 oneDNN operator

### PR DESCRIPTION
**Pull request by grygielski**
### PR types
Function optimization

### PR changes
OPs

### Describe
This PR refactors oneDNN fp32 convolution operator. It will make code more maintainable and be a base for future refactoring of other convolutions.

### CODE REVIEW

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc [hunk 1]
```diff
@@ -26,11 +26,11 @@ using mkldnn::memory;
 using mkldnn::primitive;
 using mkldnn::reorder;
 using mkldnn::stream;
-using platform::to_void_cast;
 using platform::GetMKLDNNFormat;
+using platform::to_void_cast;
 
 inline void GetWeightsTz(std::vector<int64_t>& weights_tz,  // NOLINT
-                         int groups, bool is_conv3d) {
```
Comment (lidanqing-vv) [inline paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:33]: Hi, Do you consider rewriting this function using std::rotate? It should not be slower right?

Comment (grygielski) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:33]: I will give it a try, good idea.

**Review by jczaja — APPROVED** [review]
LGTM

**Comment by wojtuss** [comment]
I have verified that this PR fixes also https://github.com/PaddlePaddle/Paddle/issues/24985

**Comment by grygielski** [comment]
@luotao1 PR_CI_CPU_Py2 needs your approval to pass

**Review by luotao1 — APPROVED** [review]

**Pull request merged**